.github/workflows/ci.yml

```
@@ -16,9 +16,30 @@ jobs:
+
+    services:
+      postgres:
+        image: postgres:16
+        env:
+          # Important: env.* does NOT work here, so we repeat the expressions
+          POSTGRES_USER: ${{ secrets.CI_POSTGRES_USER || vars.CI_POSTGRES_USER || 'devops' }}
+          POSTGRES_PASSWORD: ${{ secrets.CI_POSTGRES_PASSWORD || 'devops' }}
+          POSTGRES_DB: ${{ secrets.CI_POSTGRES_DB || vars.CI_POSTGRES_DB || 'whoknows' }}
+        ports:
+          - 5432:5432
+        options: >-
```

actionlint (1.7.9)

context "secrets" is not allowed here. available contexts are "github", "inputs", "matrix", "needs", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)